lib/common/grpc-service.js

```diff
@@ -266,6 +268,25 @@ GrpcService.prototype.request = function(protoOpts, reqOpts, callback) {
     request: function(_, onResponse) {
       respError = null;
 
+      if (stream) {
+        service[protoOpts.method](reqOpts, grpcOpts)
+          .pipe(stream)
```

Comment:
This is a bit tricky, because if the request is retried, the stream will have already received events and data. In retry-request, I use a cache stream to hold these events until a successful request has been initiated, then releasing them by piping it to the stream the user is holding: https://github.com/stephenplusplus/retry-request/blob/c9a5d8ea5eb23d504be12e1b319c287ea8401f8a/index.js#L83 (just search the page for `cacheStream` to see how it is used).

I *think* retry-request should be able to use its caching stream logic if we don't pass retry-request a callback when we are in stream mode.